examples/model_compress/pruning/pruning_bert_glue.py

```diff
@@ -0,0 +1,355 @@
+                if save_best_model and best_result < result[0]:
+                    assert save_path is not None
+                    torch.save(model.state_dict(), save_path)
+                    best_result = result[0]
+            if max_steps and current_step >= max_steps:
+                return
+
+
+def evaluation(model):
+    training = model.training
+    model.eval()
+    is_regression = task_name == 'stsb'
+    metric = load_metric('glue', task_name)
+
+    for batch in validate_dataloader1:
+        batch.to(device)
+        outputs = model(**batch)
+        predictions = outputs.logits.argmax(dim=-1) if not is_regression else outputs.logits.squeeze()
+        metric.add_batch(
+            predictions=predictions,
+            references=batch['labels'],
+        )
+    result = metric.compute()
+
+    if validate_dataloader2:
+        for batch in validate_dataloader2:
+            batch.to(device)
+            outputs = model(**batch)
+            predictions = outputs.logits.argmax(dim=-1) if not is_regression else outputs.logits.squeeze()
+            metric.add_batch(
+                predictions=predictions,
+                references=batch['labels'],
+            )
+        result = {'matched': result, 'mismatched': metric.compute()}
+
+        model.train(training)
+        return (result['matched']['accuracy'] + result['mismatched']['accuracy']) / 2, result
+
+    model.train(training)
+    return result.get('f1', result.get('accuracy', None)), result
+
+
+# using huggingface native loss
+def fake_criterion(outputs, targets):
+    return outputs
+
+
+if __name__ == '__main__':
+    ##########################################################################################################################
+    # The entire pruning process can be divided into the following steps:
+    # 1. Finetune the pretrained model on the downstream task. From our experience,
+    #    the final performance of pruning on the finetuned model is better than pruning directly on the pretrained model.
+    #    At the same time, the finetuned model obtained in this step will also be used as the teacher model for the following
+    #    distillation training.
+    # 2. Pruning the attention layer at first. Here we apply block-sparse on attention layer weight,
+    #    and directly prune the head (condense the weight) if the head was fully masked.
+    #    If the head was partial masked, we will not prune it and recover its weight.
+    # 3. Retrain the head-pruned model with distillation. Recover the model precision before pruning FFN layer.
+    # 4. Pruning the FFN layer. Here we apply the output channels pruning on the 1st FFN layer,
+    #    and the 2nd FFN layer input channels will be pruned due to the pruning of 1st layer output channels.
+    # 5. Retrain the final pruned model with distillation.
```

Comment:
how about moving this part to the doc page?

Reply:
@ultmaster thinks that users may not understand this if this is in the document, should I keep this part in both doc and example code?

Reply:
no need to keep two copies, because each step has been explained below, no need to summarize the step list at the top of the `__main__` function.

Reply:
The best approach is to write it as a tutorial. Then we can have both a python code and a HTML webpage, and we don't have to write it twice.